From fbbeab11e7c6bac4903cccfef194a6d6ac3810b2 Mon Sep 17 00:00:00 2001
From: Shan Kang <shan.kang@intel.com>
Date: Thu, 31 Oct 2019 15:26:46 +0800
Subject: [PATCH] Update lto related for v0.3.7

This is an updated patch to port Victor's lto patch to adapt to
openBLAS v0.3.7

Signed-off-by: Shan Kang <shan.kang@intel.com>
---
 Makefile.system                         | 10 +++++-----
 lapack-netlib/INSTALL/make.inc.gfortran |  6 +++---
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/Makefile.system b/Makefile.system
index 6addbda..9b36a79 100644
--- a/Makefile.system
+++ b/Makefile.system
@@ -263,10 +263,10 @@
 
 ARFLAGS	=
 CPP	= $(COMPILER) -E
-AR	?= $(CROSS_SUFFIX)ar
+AR	?= $(CROSS_SUFFIX)gcc-ar
 AS	?= $(CROSS_SUFFIX)as
 LD	?= $(CROSS_SUFFIX)ld
-RANLIB	?= $(CROSS_SUFFIX)ranlib
+RANLIB	?= $(CROSS_SUFFIX)gcc-ranlib
 NM	= $(CROSS_SUFFIX)nm
 DLLWRAP = $(CROSS_SUFFIX)dllwrap
 OBJCOPY = $(CROSS_SUFFIX)objcopy
@@ -293,7 +293,7 @@ MD5SUM = md5 -n
 endif
 
 ifeq ($(OSNAME), Linux)
-EXTRALIB	+= -lm
+EXTRALIB	+= -lm -lgfortran
 NO_EXPRECISION = 1
 endif
 
@@ -1185,11 +1185,11 @@ FCOMMON_OPT += -g
 endif
 
 ifndef COMMON_OPT
-COMMON_OPT = -O2
+COMMON_OPT = -O3
 endif
 
 ifndef FCOMMON_OPT
-FCOMMON_OPT = -O2 -frecursive
+FCOMMON_OPT = -O3 -frecursive
 endif
 
 override CFLAGS     += $(COMMON_OPT) $(CCOMMON_OPT) -I$(TOPDIR)
diff --git a/lapack-netlib/INSTALL/make.inc.gfortran b/lapack-netlib/INSTALL/make.inc.gfortran
index 39d98d4..afc83b9 100644
--- OpenBLAS-0.3.8/lapack-netlib/INSTALL/make.inc.gfortran~	2020-02-09 22:16:28.000000000 +0000
+++ OpenBLAS-0.3.8/lapack-netlib/INSTALL/make.inc.gfortran	2020-02-10 15:53:10.979388033 +0000
@@ -31,9 +31,9 @@
 #  The archiver and the flag(s) to use when building an archive
 #  (library).  If your system has no ranlib, set RANLIB = echo.
 #
-AR = ar
+AR = gcc-ar
 ARFLAGS = cr
-RANLIB = ranlib
+RANLIB = gcc-ranlib
 
 #  Timer for the SECOND and DSECND routines
 #
